src/components/storefront/CartItem/CartItem.js

```diff
@@ -1,12 +1,18 @@
 import React, { useEffect, useRef } from 'react';
-
-import { Heading, Flex, Image, useNumberInput, IconButton, VStack } from '@chakra-ui/react';
+import { Heading, Flex, Image, useNumberInput, IconButton, VStack, Link } from '@chakra-ui/react';
 import QuantityPicker from '../ProductDetails/QuantityPicker';
 import { CloseIcon } from '@chakra-ui/icons';
 import { useShopify } from 'hooks/useShopify';
+import PreserveQueryParamsLink from 'components/storefront/PreserveQueryParamsLink/PreserveQueryParamsLink';
 
 const CartItem = ({ title, sku, image, price, quantity, lineItemId, checkoutId }) => {
-  const { removeLineItem, updateQuantity } = useShopify();
+  const {
+    removeLineItem,
+    updateQuantity,
+    collections: { data },
+  } = useShopify();
+  // Retrieves the slug from "All Items" collection since id in checkout is different than id in "All Items"
+  const slug = data[2].products.find(product => product.title === title).id;
```

Comment:
why is it different — can we make it the same? or does it have to do with it being the `id` of the variant or something?

I'm sure this code here works, but it's obviously not ideal

Reply:
So the reason it's different is because inside `productDetails.js` we're passing in `variantId` as a lineItemToAdd to checkout, whereas the slug for the product details pages are based off of the id of the product. I tried replacing `variantId` with `productId` when adding it to checkout, but I get an error for an invalid id (I assume we can only add variants to checkout, rather than products). 
![image](https://user-images.githubusercontent.com/24748963/115977023-a6831d80-a541-11eb-8116-4e102725251e.png)

![image](https://user-images.githubusercontent.com/24748963/115977019-99662e80-a541-11eb-9715-81c3c22fcc27.png)

not entirely sure what to do about this

Reply:
decided to use `products` and leave this as is (added in a loading check for `productsLoading` inside `Cart.js`